test/conformance/service_test.go

```diff
@@ -536,4 +536,47 @@ func TestReleaseService(t *testing.T) {
 	}
 }
 
+func TestMultipleNamespace(t *testing.T) {
+	t.Parallel()
+
+	altServiceNamespace := fmt.Sprintf("%s-%s", test.ServingNamespace, test.ObjectNameForTest(t))
+	serviceName := test.ObjectNameForTest(t)
+
+	defaultClients := e2e.Setup(t) // This one uses the default namespace `test.ServingNamespace`
+	altClients := e2e.SetupWithNamespace(t, altServiceNamespace)
+
+	_, err := test.CreateNamespace(altClients, altServiceNamespace)
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer test.DeleteNamespace(altClients, altServiceNamespace)
+
+	defaultResources := test.ResourceNames{
+		Service: serviceName,
+		Image:   pizzaPlanet1,
+	}
+	if _, err := test.CreateRunLatestServiceReady(t, defaultClients, &defaultResources, &test.Options{}); err != nil {
+		t.Fatalf("Failed to create Service %v in namespace %v: %v", defaultResources.Service, defaultResources.Namespace, err)
+	}
+	defer test.TearDown(defaultClients, defaultResources)
+
+	altResources := test.ResourceNames{
+		Service: serviceName,
+		Image:   pizzaPlanet2,
+		Namespace: altServiceNamespace,
```

Comment:
Why create an use an 'altClient' if you explicitly set the Namespace on the 'altResource'? It seems like you could use the same client and explicitly set namespace on both?

Reply:
https://github.com/knative/serving/blob/5dc9a1f850129e5017b6ecb6cc816724e31aff87/pkg/client/clientset/versioned/typed/serving/v1alpha1/service.go#L97

It uses the namespace that was initially defined when creating the service. When using the default test client, it ends up explicitly requesting for the namespace `serving-tests` while the actual service declaration is asking for `serving-tests-multiple-namespace-xxxxx`. This results in it complaining about a mismatch.

Reply:
Thinking about this further if we are going to keep this in conformance testing I would like to move namespace creation outside of this method and into setup like we do for the base `serving-tests` namespace. 

Even though it is only for setup the kubeclient is used and executed in the context of this test. This really goes against the 4th requirement ( https://github.com/knative/serving/tree/master/test/conformance#requirements ) of our conformance tests.

I think this could be updated by either moving this to e2e tests OR changing to a fixed namespace that is setup in `e2e-common.sh`, configurable in `e2e_flags.go` and documented in `README.md`

Relevant places:
* https://github.com/knative/serving/tree/master/test#environment-requirements
* https://github.com/knative/serving/blob/bbe4c2f6e2d383130a6fb9a802c456a04ec27fc0/test/e2e-common.sh#L220
* https://github.com/knative/serving/blob/32499ed908cdbab6067ac242814e9391e5c4c30a/test/e2e_flags.go#L31

Reply:
Fair points, I shall update to one of the suggested options

Reply:
I've moved it to an e2e test.